websocket-chat/Sources/App/ConnectionManager.swift

```diff
@@ -68,36 +78,67 @@ struct ConnectionManager: Service {
     }
 
     func run() async {
+        /// The `withGracefulShutdownHandler` is a helper that will call the `onGracefulShutdown` closure
+        /// when the application is shutting down.
+        /// 
+        /// This helps ensure that the application will not exit before the connection manager has a chance to
+        /// clean up all the connections.
         await withGracefulShutdownHandler {
+            /// The `withDiscardingTaskGroup` is a task group that can indefinitely add tasks to it.
+            /// As opposed to a regular task group, it will not incur memory overhead for each task added.
+            /// This allows it to scale for a large number of tasks.
             await withDiscardingTaskGroup { group in
+                // The OutboundConnections actor is used to manage the outbound connections in a thread safe manner
+                // Allowing us to broadcast messages to all the connected clients
                 let outboundCounnections = OutboundConnections()
+
+                // As each client connects, the for loop will emit the next connection
                 for await connection in self.connectionStream {
+                    // Each client connection is handled in a new task, so their work is parallelized
                     group.addTask {
                         self.logger.info("add connection", metadata: ["name": .string(connection.name)])
+
+                        // Add the client to the list of connected clients
                         guard await outboundCounnections.add(name: connection.name, outbound: connection.outbound) else {
+                            // If the client already exists, we close the connection and return out of the Task
                             self.logger.info("user already exists", metadata: ["name": .string(connection.name)])
                             await connection.outbound.send(.close("User connected already"))
                             connection.outbound.finish()
                             return
                         }
 
                         do {
+                            // We handle the stream is incoming messages emitted by this client
```

Comment:
Doesn't make complete sense. Did you mean `as`?
```suggestion
                            // We handle the stream as incoming messages emitted by this client
```